lib/irslackd.js

```diff
@@ -299,13 +310,121 @@ class Irslackd {
   }
   async onIrcPing(ircUser, msg) {
     // Send PONG
-    this.ircd.write(ircUser.socket, 'irslackd', 'PONG', [ 'irslackd' ]);
+    this.ircd.write(ircUser.socket, 'irslackd', 'PONG', ['irslackd']);
   }
   async onIrcMode(ircUser, msg) {
     let target = msg.args[0];
     if (target.substr(0, 1) === '#') {
-      this.ircd.write(ircUser.socket, 'irslackd', 'MODE', [ target ]);
+      this.ircd.write(ircUser.socket, 'irslackd', 'MODE', msg.args);
+    }
+  }
+  async onIrcInvite(ircUser, msg) {
+    const self = this;
+    try {
+      if (msg.args.length !== 2) {
+        self.ircd.write(ircUser.socket, 'irslackd', '371', [ircUser.ircNick, ':Must pass two params, the user to invite and the channel']);
+        return; // must have two params (nick and channel)
+      }
+      var userId = await self.lookupUserId(ircUser, msg.args[0].substr(0, 1) === '@' ? msg.args[0].substr(1) : msg.args[0]); // remove at-sign
+      var channelId = await self.lookupChannelId(ircUser, msg.args[1].substr(0, 1) === '#' ? msg.args[1].substr(1) : msg.args[1]); // remove hash
```

Comment:
```js
    const userId = ircUser.ircToSlack.get(msg.args[0]);
    const channelId = ircUser.ircToSlack.get(msg.args[1]);
    if (!userId) throw Error("IRC nick not found");
    if (!channelId) throw Error("IRC channel not found ");
```

Reply:
oh, nice ... didn't realize a map of ids existed already ... will look

Reply:
i found the ircToSlack map .... what's the name of the map for channelids?

Reply:
`ircToSlack` contains both nicks and channels. Channels are prefixed with `#`.

I intend on making separate maps for each target type to solve #46 but for now we can just use that.

Reply:
roger

Reply:
are there any users and channels loaded into ircToSlack map for testing purposes?

Reply:
i manually loaded user and channel in my test, let me know if that's incorrect